File: VB/WindowsFormsApplication39/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraBars.Controls

Namespace WindowsFormsApplication39
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            AddHandler Me.MouseMove, AddressOf Form1_MouseMove
            AddHandler Me.Paint, AddressOf Form1_Paint
            AddHandler bar1.VisibleChanged, AddressOf bar1_VisibleChanged

            bar1.Visible = False
        End Sub

        Private Sub bar1_VisibleChanged(ByVal sender As Object, ByVal e As EventArgs)
            If bar1.Visible Then
                Dim control As CustomBarControl = bar1.GetBarControl()
                AddHandler control.MouseLeave, AddressOf control_MouseLeave
            End If
        End Sub

        Private Sub Form1_MouseMove(ByVal sender As Object, ByVal e As MouseEventArgs)
            Dim rect As Rectangle = Me.ClientRectangle
            rect.Width = 20
            If rect.Contains(e.Location) Then
                ShowBar()
            End If
        End Sub
        Private Sub control_MouseLeave(ByVal sender As Object, ByVal e As EventArgs)
            HideBar()
        End Sub

        Private Sub ShowBar()
            bar1.Visible = True
        End Sub

        Private Sub HideBar()
            bar1.Visible = False
        End Sub

        Private Sub Form1_Paint(ByVal sender As Object, ByVal e As PaintEventArgs)
            Dim rect As Rectangle = Me.ClientRectangle
            rect.Width = 20
            e.Graphics.FillRectangle(Brushes.AliceBlue, rect)

            Dim drawString As String = "Hover this region to show Bar"

            Dim drawBrush As New System.Drawing.SolidBrush(System.Drawing.Color.Black)
            Dim drawFormat As New System.Drawing.StringFormat(StringFormatFlags.DirectionVertical)
            e.Graphics.DrawString(drawString, Me.Font, drawBrush, 3, 3, drawFormat)
        End Sub
    End Class
End Namespace

File: CS/WindowsFormsApplication39/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars.Controls;

namespace WindowsFormsApplication39 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            this.MouseMove += Form1_MouseMove;
            this.Paint += Form1_Paint;
            bar1.VisibleChanged += bar1_VisibleChanged;

            bar1.Visible = false;
        }

        void bar1_VisibleChanged(object sender, EventArgs e) {
            if(bar1.Visible) {
                CustomBarControl control = bar1.GetBarControl();
                control.MouseLeave += control_MouseLeave;
            }
        }

        void Form1_MouseMove(object sender, MouseEventArgs e) {
            Rectangle rect = this.ClientRectangle;
            rect.Width = 20;
            if(rect.Contains(e.Location)) {
                ShowBar();
            }
        }
        void control_MouseLeave(object sender, EventArgs e) {
            HideBar();
        }

        private void ShowBar() {
            bar1.Visible = true;
        }

        private void HideBar() {
            bar1.Visible = false;
        }

        void Form1_Paint(object sender, PaintEventArgs e) {
            Rectangle rect = this.ClientRectangle;
            rect.Width = 20;
            e.Graphics.FillRectangle(Brushes.AliceBlue, rect);

            string drawString = "Hover this region to show Bar";

            System.Drawing.SolidBrush drawBrush = new System.Drawing.SolidBrush(System.Drawing.Color.Black);
            System.Drawing.StringFormat drawFormat = new System.Drawing.StringFormat(StringFormatFlags.DirectionVertical);
            e.Graphics.DrawString(drawString, this.Font, drawBrush, 3, 3, drawFormat);
        }
    }
}
